### PR TITLE
Install event hook for environment and configureEnvironment options

### PR DESCRIPTION
**Pull request by Randshot**
This PR addresses issue #346.

### CODE REVIEW

File: src/cms-driver.ts [hunk 1]
```diff
@@ -5,6 +5,7 @@ import * as api from './api';
 import {CacheEntryProperties, ExecutableTarget, RichTarget} from './api';
 import * as cache from './cache';
 import * as cms from './cms-client';
+import cfg from './config';
```
Comment (vector-of-bool) [inline src/cms-driver.ts]: Can you change `cfg` -> `config`, just to match the rest of the project.

Comment (Randshot) [reply src/cms-driver.ts]: Then I will have to change every local varname with the `config` identifier. There are a couple of them in the `cms-driver.ts` file.
@vector-of-bool Should I do that?

Comment (vector-of-bool) [reply src/cms-driver.ts]: Go for it.

Comment (Randshot) [reply src/cms-driver.ts]: Done.

**Pull request merged**